Remotion/ObjectBinding/Web.UnitTests/UI/Controls/Validation/BocListValidationResultDispatchingValidatorTest.cs

```diff
@@ -0,0 +1,174 @@
+
+      var businessObjectStub = new Mock<IBusinessObject>();
+
+      var bocListMock = new Mock<Control> { CallBase = true };
+      bocListMock.As<IBocList>().Setup(_ => _.ValidationFailureRepository).Returns(validationFailureRepoStub.Object);
+      bocListMock.As<IBocList>().Setup(_ => _.GetResourceManager()).Returns(resourceManagerStub.Object);
+      bocListMock.As<IBocList>().Setup(_ => _.Value).Returns(new[] { businessObjectStub.Object });
+      bocListMock.Object.ID = "someBocList";
+
+      var validator = new BocListValidationResultDispatchingValidator();
+      validator.ErrorMessage = "This should not be output.";
+      validator.ControlToValidate = "someBocList";
+
+      NamingContainer.Controls.Add(bocListMock.Object);
+      NamingContainer.Controls.Add(validator);
+
+      ((IValidatorWithDynamicErrorMessage)validator).RefreshErrorMessage();
+
+      Assert.That(validator.ErrorMessage, Is.EqualTo("Errors on rows in BocList\r\n"));
+    }
+
+    [Test]
+    public void RefreshErrorMessage_WithInternalAndRowFailures_ReturnsAppendedMessageText ()
+    {
+      var validationFailuresForBocList = new[]
+                                         {
+                                           BocListValidationFailureWithLocationInformation.CreateFailure(BusinessObjectValidationFailure.Create("Errors on BocList"))
+                                         };
+
+      var validationFailuresForBocListRows = new[]
+                                             {
+                                               BocListValidationFailureWithLocationInformation.CreateFailure(BusinessObjectValidationFailure.Create("Errors on BocList rows"))
+                                             };
+
+      var validationFailureRepoStub = new Mock<IBocListValidationFailureRepository>();
+      validationFailureRepoStub.Setup(_ => _.GetUnhandledValidationFailuresForBocList(true)).Returns(validationFailuresForBocList);
+      validationFailureRepoStub.Setup(_ => _.GetUnhandledValidationFailuresForBocListAndContainingDataRowsAndDataCells(true)).Returns(validationFailuresForBocListRows);
+      var error = "Errors on rows in BocList";
+
+      var resourceManagerStub = new Mock<IResourceManager>();
+      resourceManagerStub.Setup(_ => _.TryGetString("Remotion.ObjectBinding.Web.UI.Controls.BocList.RemainingValidationFailureText", out error)).Returns(true);
+
+      var businessObjectStub = new Mock<IBusinessObject>();
+
+      var bocListMock = new Mock<Control> { CallBase = true };
+      bocListMock.As<IBocList>().Setup(_ => _.ValidationFailureRepository).Returns(validationFailureRepoStub.Object);
+      bocListMock.As<IBocList>().Setup(_ => _.GetResourceManager()).Returns(resourceManagerStub.Object);
+      bocListMock.As<IBocList>().Setup(_ => _.Value).Returns(new[] { businessObjectStub.Object });
+      bocListMock.Object.ID = "someBocList";
+
+
+
+      var validator = new BocListValidationResultDispatchingValidator();
```

Comment:
two lines is okay, three is a bit much. And for symetry with the rest of the tests, one is enough. otherwise, we would need two lines in all the tests here.
```suggestion
      bocListMock.Object.ID = "someBocList";

      var validator = new BocListValidationResultDispatchingValidator();
```